projects/fw_103/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include <stdint.h>  // for integer types
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"  // interrupts are required for soft timers
+#include "log.h"        // for printing
+#include "wait.h"       // for wait function
+
+GpioAddress adc_addr = {
+  .port = GPIO_PORT_A,
+  .pin = 6,
+};
+
+static void prv_read_adc(const GpioAddress *btn_addr_ptr, void *context) {
+  uint16_t adc_data = 0;
+  if (adc_read_converted_pin(adc_addr, &adc_data) == STATUS_CODE_OK) {
+    LOG_DEBUG("ADC data: %d\n", adc_data);
+  } else {
+    LOG_DEBUG("Failed to read adc data");
+  }
+}
+
+int main(void) {
+  gpio_init();
+  gpio_it_init();
+  interrupt_init();
+
+  GpioAddress btn_addr = {
+    .port = GPIO_PORT_B,
+    .pin = 2,
+  };
+  GpioSettings adc_gpio_settings = {
+    .direction = GPIO_DIR_IN,
+    .state = GPIO_STATE_LOW,
+    .resistor = GPIO_RES_NONE,
+    .alt_function = GPIO_ALTFN_ANALOG,
+  };
+  GpioSettings btn_gpio_settings = {
+    .direction = GPIO_DIR_IN,
+    .state = GPIO_STATE_LOW,
+    .resistor = GPIO_RES_PULLDOWN,
```

Comment:
I think this should be non instead of pulldown? but tbh, I'm not exactly sure myself.

Reply:
I did not have a reference schematic, and so assumed an internal pull-down resistor. But `None` would also work for a external pull-down. I'll update this.